game-core/src/main/java/org/triplea/game/server/AvailableGames.java

```diff
@@ -35,19 +36,22 @@
   private final Map<String, URI> availableGames;
 
   AvailableGames() {
-    availableGames = Collections.synchronizedMap(new HashMap<>());
-
-    FileUtils.listFiles(ClientFileSystemHelper.getUserMapsFolder())
-        .parallelStream()
-        .forEach(
-            map -> {
-              log.info("Loading map: " + map);
-              if (map.isDirectory()) {
-                availableGames.putAll(getGamesFromDirectory(map));
-              } else if (map.isFile() && map.getName().toLowerCase().endsWith(ZIP_EXTENSION)) {
-                availableGames.putAll(getGamesFromZip(map));
-              }
-            });
+    availableGames =
+        FileUtils.listFiles(ClientFileSystemHelper.getUserMapsFolder())
+            .parallelStream()
+            .<Map<String, URI>>map(
+                map -> {
```

Comment:
nit, have you considered extracting this to a function?

Often streams are best when there is no or little body (eg; try to convert most of the blocks into method references).

Beyond that, looks fine, feel free to update and merge. (I'll be AFK for a day or two).

Reply:
@RoiEXLab I"m not sure how much we really moved the needle on this one by getting rid of the `forEach`, in some ways the extra syntax introduced makes it perhaps only very marginal of an improvement:

```

            .<Map<String, URI>>map(
        .forEach(
                map -> {
	
```

I do think the big improvement would be to convert the lambda blocks to be method references. Though, at the end of the day having tests drive the design is probably even better.

Reply:
Ah, sorry for not replying, I wanted to extract it to a method (would also get rid of the explicit type parameters), but I simply just haven't got to it yet.

Will follow up soon-ish